src/test/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessCategoriesServiceTest.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.google.common.collect.Sets;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CaseAccessCategoriesServiceTest {
+
+    private CaseAccessCategoriesService caseAccessCategoriesService;
+
+    @BeforeEach
+    void setUp() {
+        caseAccessCategoriesService = new CaseAccessCategoriesService();
+    }
+
+    @Test
+    void shouldReturnTrueWhenCreateCase() {
+        AccessProfile accessProfile = mock(AccessProfile.class);
+        Predicate<CaseDetails> caseDetailsPredicate = caseAccessCategoriesService
+            .caseHasMatchingCaseAccessCategories(Sets.newHashSet(accessProfile), true);
+        CaseDetails caseDetails = new CaseDetails();
+        assertTrue(caseDetailsPredicate.test(caseDetails));
+    }
+
+    @Test
+    void shouldReturnTrueWhenAccessProfilesAreEmpty() {
```

Comment:
If there are no access profiles, I think we should return `false`? Presumably access control elsewhere would always prevent access anyway, but better to be safe!